package/MDAnalysis/coordinates/H5MD.py

```diff
@@ -0,0 +1,587 @@
+
+    def close(self):
+        """close reader"""
+        if self._driver != 'mpio':
+            self._file.close()
+
+    def _reopen(self):
+        """reopen trajectory
+
+        Note
+        ----
+        This method does close the and reopen the file like most other
+        coordinate readers, as this would cause problems with reopening
+        the file with the ``driver`` and ``comm`` arguments for parallel
+        reads. Instead, it rewinds the trajectory back to the first timstep
+        if 'mpio' is passed for the driver.
+
+        """
+        if self._driver != 'mpio':
+            self.close()
+            self.open_trajectory()
+        else:
+            self._frame = -2
+            self._read_next_timestep()
+
+    @property
+    def has_positions(self):
+        """True if 'position' group is in trajectory."""
+        return self._has['position']
+
+    @has_positions.setter
+    def has_positions(self, value: bool):
+        self._has['position'] = value
+
+    @property
+    def has_velocities(self):
+        """True if 'velocity' group is in trajectory."""
+        return self._has['velocity']
+
+    @has_velocities.setter
+    def has_velocities(self, value: bool):
+        self._has['velocity'] = value
+
+    @property
+    def has_forces(self):
+        """True if 'force' group is in trajectory."""
+        return self._has['force']
+
+    @has_forces.setter
+    def has_forces(self, value: bool):
+        self._has['force'] = value
+
+    @property
+    def n_frames(self):
+        """number of frames in trajectory"""
+        for name, value in self._has.items():
+            if value:
+                return self._particle_group[name]['value'].shape[0]
+
+    def _read_frame(self, frame):
+        """reads data from h5md file and copies to current timestep"""
+        try:
+            for name, value in self._has.items():
+                if value:
+                    myframe = self._particle_group[name]['step'][frame]
+                    break
+            else:
+                raise NoDataError("Provide at least a position, velocity"
+                                  " or force group in the h5md file.")
+        except ValueError:
+            raise IOError from None
+
+        self._frame = frame
+        ts = self.ts
+        particle_group = self._particle_group
+        ts.frame = frame
+
+        # fills data dictionary from 'observables' group
+        # Note: dt is not read into data as it is not decided whether
+        # Timestep should have a dt attribute (see Issue #2825)
+        self._copy_to_data()
+
+
+        # Sets frame box dimensions
+        # Note: H5MD files must contain 'box' group in each particle group
+        if 'edges' in particle_group['box']:
+            ts._unitcell[:] = particle_group['box/edges/value'][frame, :]
+        else:
+            # sets ts.dimensions = [0, 0, 0, 0, 0, 0]
+            ts._unitcell[:] = None
```

Comment:
Does this work? Can you fill a float32 array with `None` values??

Reply:
The assignment is allowed, but I'm not sure if `None` vs `nan` is an issue...
![image](https://user-images.githubusercontent.com/59893899/89116425-c5d49a80-d448-11ea-95af-82630b5b4ee6.png)
and it _seems_ to work in that this test I wrote passes:
https://github.com/MDAnalysis/mdanalysis/blob/d408d94f34c1345beb6439dbf90bc382a672349c/testsuite/MDAnalysisTests/coordinates/test_h5md.py#L397-L407
but I didn't check explicitly.